crates/circuit/src/lib.rs

```diff
@@ -37,53 +37,60 @@ use pyo3::prelude::*;
 use pyo3::types::{PySequence, PyTuple};
 use pyo3::PyTypeInfo;
 
-pub type BitType = u32;
 #[derive(Copy, Clone, Debug, Hash, Ord, PartialOrd, Eq, PartialEq, FromPyObject)]
-pub struct Qubit(pub BitType);
+pub struct Qubit(pub u32);
 
-impl Qubit {
-    /// Construct a new Qubit object from a usize, if you have a u32 you can
-    /// create a `Qubit` object directly with `Qubit(0u32)`. This will panic
-    /// if the `usize` index exceeds `u32::MAX`.
-    #[inline(always)]
-    pub fn new(index: usize) -> Self {
-        Qubit(
-            index.try_into().unwrap_or_else(|_| {
-                panic!("Index value '{}' exceeds the maximum bit width!", index)
-            }),
-        )
-    }
-
-    /// Convert a Qubit to a usize
-    #[inline(always)]
-    pub fn index(&self) -> usize {
-        self.0 as usize
-    }
-}
+#[derive(Copy, Clone, Debug, Hash, Ord, PartialOrd, Eq, PartialEq, FromPyObject)]
+pub struct Clbit(pub u32);
+
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq)]
+pub struct Var(u32);
+
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq)]
+pub struct Stretch(u32);
```

Comment:
Is there a reason to not make the inner u32 pub here?

Reply:
I didn't make it public since it doesn't have real meaning, at least compared to the int of a `Qubit` or `Clbit`.